File: rust/src/entities/prohibition.rs
use serde::{Deserialize, Serialize};

use super::{action::Action, constraint::Constraint};

#[derive(Clone, Deserialize, Serialize)]
pub struct Prohibition {
    pub assignee: Option<String>,
    pub assigner: Option<String>,
    pub target: Option<String>,
    pub uid: Option<String>,
    pub constraints: Option<Vec<Constraint>>,
    pub action: Option<Action>,
}

File: src/entities/policy/prohibition.ts
import { Action } from "./action";
import { Constraint } from "./constraint";
import { Duty } from "./duty";

export interface Prohibition {
  assignee?: string;
  assigner?: string;
  target?: string;
  uid?: string;
  constraint?: Constraint[];
  action?: Action | string;
  remedies?: Duty[];
}
